src/Commands/core/ArchiveRestoreCommands.php

```diff
@@ -0,0 +1,718 @@
+{
+    use SiteAliasManagerAwareTrait;
+
+    /**
+     * @var \Symfony\Component\Filesystem\Filesystem
+     */
+    private Filesystem $filesystem;
+
+    /**
+     * @var array
+     */
+    private array $siteStatus;
+
+    /**
+     * @var null|string
+     */
+    private ?string $destinationPath = null;
+
+    private const COMPONENT_CODE = 'code';
+
+    private const COMPONENT_FILES = 'files';
+
+    private const COMPONENT_DATABASE = 'database';
+    private const SQL_DUMP_FILE_NAME = 'database.sql';
+    private const SITE_SUBDIR = 'default';
+
+    private const TEMP_DIR_NAME = 'uncompressed';
+
+    /**
+     * Restore (import) your code, files, and database.
+     *
+     * @command archive:restore
+     * @validate-php-extension Phar
+     * @aliases arr
+     *
+     * @option destination-path The base path to restore the code/files into.
+     * @option overwrite Overwrite files if exists when un-compressing an archive.
+     * @option site-subdir Site subdirectory to put settings.local.php into.
+     * @option setup-database-connection Sets up the database connection in settings.local.php file if either --db-url option or set of specific --db-* options are provided.
+     * @option code Import code.
+     * @option code-source-path Import code from specified directory. Has higher priority over "path" argument.
+     * @option code-no-composer-install Skip installing Composer dependencies.
+     * @option code-composer-install-timeout Timeout for `composer install` execution.
+     * @option files Import Drupal files.
+     * @option files-source-path Import Drupal files from specified directory. Has higher priority over "path" argument.
+     * @option files-destination-relative-path Import Drupal files into specified directory relative to Composer root.
+     * @option db Import database.
+     * @option db-source-path Import database from specified dump file. Has higher priority over "path" argument.
+     * @option db-name Destination database name.
+     * @option db-port Destination database port.
+     * @option db-host Destination database host.
+     * @option db-user Destination database user.
+     * @option db-password Destination database user password.
+     * @option db-prefix Destination database prefix.
+     * @option db-driver Destination database driver.
+     *
+     * @optionset_sql
+     * @optionset_table_selection
+     *
+     * @bootstrap none
+     *
+     * @param string|null $path
+     *   The full path to a single archive file (*.tar.gz) or a directory with components to import.
+     *   May contain the following components generated by `archive:dump` command:
+     *   1) code ("code" directory);
+     *   2) database dump file ("database/database.sql" file);
+     *   3) Drupal files ("files" directory).
+     * @param string|null $site
+     *   Destination site alias. Defaults to @self.
+     * @param array $options
+     *
+     * @throws \Exception
+     */
+    public function restore(
```

Comment:
Lets add a validate hook which checks `!elf::programExists('rsync)` and throws an exception if untrue.